Remotion/Data/DomainObjects.UnitTests/Persistence/Rdbms/SqlServer/SqlDialectTest.cs

```diff
@@ -49,5 +57,219 @@ public void GetParameterName ()
       Assert.That(_dialect.GetParameterName("parameter"), Is.EqualTo("@parameter"));
       Assert.That(_dialect.GetParameterName("@parameter"), Is.EqualTo("@parameter"));
     }
+
+    [Test]
+    [CLSCompliant(false)]
+    [TestCase(null, TestName = "CreateDataParameter_WithoutSize_DoesNotSetSizeOnParameter.")]
+    [TestCase(-1, TestName = "CreateDataParameter_WithNegativeSize_SetsSizeOnParameter.")]
+    [TestCase(0, TestName = "CreateDataParameter_WithSizeZero_SetsSizeOnParameter.")]
+    [TestCase(1, TestName = "CreateDataParameter_WithPositiveSize_SetsSizeOnParameter.")]
+    public void CreateDataParameter (int? storageTypeSize)
+    {
+      var commandMock = new Mock<IDbCommand>(MockBehavior.Strict);
+      var dataParameterMock = new Mock<IDbDataParameter>(MockBehavior.Strict);
+
+      var storageTypeInformation = new StorageTypeInformation(
+          typeof(bool),
+          "test",
+          DbType.Boolean,
+          false,
+          storageTypeSize,
+          typeof(int),
+          _typeConverterStub.Object);
+
+      _typeConverterStub.Setup(_ => _.ConvertTo(null, null, "value", storageTypeInformation.StorageType)).Returns("");
+
+      commandMock.Setup(_ => _.CreateParameter()).Returns(dataParameterMock.Object).Verifiable();
+
+      dataParameterMock.SetupSet(_ => _.ParameterName = "@dummy").Verifiable();
+      dataParameterMock.SetupSet(_ => _.DbType = storageTypeInformation.StorageDbType).Verifiable();
+      dataParameterMock.SetupSet(_ => _.Value = "").Verifiable();
+      if (storageTypeSize.HasValue)
+        dataParameterMock.SetupSet(_ => _.Size = storageTypeSize.Value).Verifiable();
+
+      var result = _dialect.CreateDataParameter(commandMock.Object, storageTypeInformation, "@dummy", "value");
+
+      commandMock.Verify();
+      dataParameterMock.Verify();
+
+      Assert.That(result, Is.SameAs(dataParameterMock.Object));
+      if (!storageTypeSize.HasValue)
+        dataParameterMock.VerifySet(_ => _.Size = It.IsAny<int>(), Times.Never());
```

Comment:
We got a strict mock above. we don't need to verify-never and do a strict mock. We can have either a stub and just verify the set at the end (or make it property behavior and just assert the properties at the end of the test) or do a strict mock and not set up the stuff we want to fail.

Right now, this last line will never happen because eitehr it's not expected to happen and then it won't get asserted or it would fail when called due to the strict-behavior.

Reply:
Reading more into the strict mock: since we can't actually test the correct value of Size when it's not been set (the default being 0), we do would be doing a VerifySet at the end. We can still do regular Assert for the other properties if we want to. Either way, no need to be strict and Verify. One or the other suffices.